### ISSUE TITLE
issue with managment server installation in using advanced folder in devcloud4

### ISSUE THREAD

**Issue by Hanan100** [imduffy15/devcloud4#11]
I am still having issues with the installation of the management server using the advanced folder (Not the binary instillation advanced). The error message I am getting is as follow

==> management: [2015-01-12T14:02:51+00:00] INFO: *** Chef 12.0.3 ***
==> management: [2015-01-12T14:02:51+00:00] INFO: Chef-client pid: 2972
==> management: [2015-01-12T14:02:54+00:00] INFO: Setting the run_list to ["recipe[cloudstack::default]"] from CLI options
==> management: [2015-01-12T14:02:54+00:00] INFO: Run List is [recipe[cloudstack::default]]
==> management: [2015-01-12T14:02:54+00:00] INFO: Run List expands to [cloudstack::default]
==> management: [2015-01-12T14:02:54+00:00] INFO: Starting Chef Run for localhost
==> management: [2015-01-12T14:02:54+00:00] INFO: Running start handlers
==> management: [2015-01-12T14:02:54+00:00] INFO: Start handlers complete.
==> management: [2015-01-12T14:02:54+00:00] ERROR: Running exception handlers
==> management: [2015-01-12T14:02:54+00:00] ERROR: Exception handlers complete
==> management: [2015-01-12T14:02:54+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> management: [2015-01-12T14:02:54+00:00] ERROR: Cookbook cloudstack not found. If you're loading cloudstack from another cookbook, make sure you configure the dependency in your metadata
==> management: [2015-01-12T14:02:54+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the

**Comment by imduffy15** [comment]
I'm unable to recreate this issue... I have set a hard coded value for the chef version that is installed onto the VM. This may or may not fix the issue... I don't know as I  am unable to recreate it.

For me the master branch is currently working perfectly. I just finished a vagrant up of the binary-installation-basic

![screen shot 2015-01-14 at 05 25 24](https://cloud.githubusercontent.com/assets/1243435/5734457/cb1c232a-9bad-11e4-8756-e9341e391896.png)

$ vagrant --version
Vagrant 1.7.2

$ vagrant plugin list
vagrant-berkshelf (4.0.2)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.3, system)

$ berks --version
3.2.1

$ vagrant ssh management
Last login: Wed Jan 14 05:26:38 2015 from 10.0.2.2
[vagrant@localhost ~]$ chef-client --version
Chef: 11.16.4

**Comment by Hanan100** [comment]
My environment settings are as follow
Machine OS : ubuntu 14.04
Virtualbox  4.3.6 r91406
vagrant plugin list
vagrant-berkshelf (4.0.2)
vagrant-librarian-chef (0.2.1)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.4, system)

berks -v
3.2.1

vagrant -v
Vagrant 1.7.2

I am still getting the same error message with regard to the advanced file (Not the binary installation advanced) Where it is not locating the cloudstack cookbook. Would it be a configuration problem with berkself ????

When attempting to use binary installation basic I receive the following error message

    xenserver: Warning: Connection timeout. Retrying...
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'starting, running'. The machine is in the
'aborted' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

**Comment by Hanan100** [comment]
The same error occurred with basic (Not the binary basic installation) 

==> management: [2015-01-14T07:41:15+00:00] INFO: *** Chef 11.16.0 ***
==> management: [2015-01-14T07:41:15+00:00] INFO: Chef-client pid: 2950
==> management: [2015-01-14T07:41:18+00:00] INFO: Setting the run_list to ["recipe[cloudstack::default]"] from CLI options
==> management: [2015-01-14T07:41:18+00:00] INFO: Run List is [recipe[cloudstack::default]]
==> management: [2015-01-14T07:41:18+00:00] INFO: Run List expands to [cloudstack::default]
==> management: [2015-01-14T07:41:18+00:00] INFO: Starting Chef Run for localhost
==> management: [2015-01-14T07:41:18+00:00] INFO: Running start handlers
==> management: [2015-01-14T07:41:18+00:00] INFO: Start handlers complete.
==> management: [2015-01-14T07:41:18+00:00] ERROR: Running exception handlers
==> management: [2015-01-14T07:41:18+00:00] ERROR: Exception handlers complete
==> management: [2015-01-14T07:41:18+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> management: [2015-01-14T07:41:18+00:00] ERROR: Cookbook cloudstack not found. If you're loading cloudstack from another cookbook, make sure you configure the dependency in your metadata
==> management: [2015-01-14T07:41:18+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

**Comment by Hanan100** [comment]
First of all thank you for your help, I truly appreciate it 
As the issue is that somehow the cloudstack cookbook is not being found, i was wondering if you can help me trace my steps to identify the cause of this error. 

My environment settings are as follow:
Machine OS : ubuntu 14.04
Virtualbox 4.3.6 r91406

 vagrant plugin list
chef-zero (3.2.1)
vagrant-berkshelf (4.0.2)
vagrant-librarian-chef (0.2.1)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.4, system)
vagrant-vbguest (0.10.0)

 berks -v
3.2.3

vagrant -v
Vagrant 1.7.2

I am using the master branch of devcloud 4

After performing the required installations I Ensured that the Chef-DK is added to the front of the path by typing the following: $ PATH=$HOME/.chefdk/gem/ruby/2.1.0/bin:/opt/chefdk/bin:$PATH

should I use ( berks init .) or (berks cookbook myapp) commands on the cookbook folder of devcloud 4

The source of these settings is http://berkshelf.com/

is there anything specific I need to do in order for berkshelf to identify the cloudstack cookbook.

waiting for your reply

**Comment by imduffy15** [comment]
Before diving into this.... is #10 resolved and working on your machine now? That is, binary-install-basic and binary-install-advanced?

**Comment by Hanan100** [comment]
It seems there is another issue with the binary installation (advanced and basic) I am receiving the following error message

  xenserver: Adapter 2: hostonly
    xenserver: Adapter 3: hostonly
    xenserver: Adapter 4: hostonly
==> xenserver: Forwarding ports...
    xenserver: 22 => 2222 (adapter 1)
==> xenserver: Running 'pre-boot' VM customizations...
==> xenserver: Booting VM...
==> xenserver: Waiting for machine to boot. This may take a few minutes...
    xenserver: SSH address: 127.0.0.1:2222
    xenserver: SSH username: vagrant
    xenserver: SSH auth method: private key
    xenserver: Warning: Connection timeout. Retrying...
The guest machine entered an invalid state while waiting for it
to boot. Valid states are 'starting, running'. The machine is in the
'aborted' state. Please verify everything is configured
properly and try again.

If the provider you're using has a GUI that comes with it,
it is often helpful to open that and watch the machine, since the
GUI often has more helpful error messages than Vagrant can retrieve.
For example, if you're using VirtualBox, run `vagrant up` while the
VirtualBox GUI is open.

**Comment by imduffy15** [comment]
And what do you see when you open virtualbox?

**Comment by Hanan100** [comment]
What appears in virtualbox a created machine with an aborted status 
![screenshot from 2015-01-17 00 05 44](https://cloud.githubusercontent.com/assets/9884801/5783474/c1892c6e-9ddc-11e4-941c-721ae46cc8bc.png)

**Comment by imduffy15** [comment]
Okay try boot up those machines in virtual box and see what happens

**Comment by Hanan100** [comment]
When I boot one of these machines my laptop gets unresponsive then after a while the booted vm goes into an aborted state again

**Comment by imduffy15** [comment]
How much ram does your laptop have?

**Comment by Hanan100** [comment]
My laptop has 6 GB of RAM and an intel core i3 processor. I was able to run the xenserver and the management server on this laptop before but now it is not working any more

**Comment by imduffy15** [comment]
OK, that explains a lot. The default vagrant configuration is 2gb for management and 6gb for xenserver.

Your machine is most likely swapping like crazy. Lets see about lowering the ram, please note you won't be able to boot any decent VMs from within cloudstack on such a low spec machine. I recommend 16gb of ram for this kind of work.

Change:
```
v.customize ['modifyvm', :id, '--memory', 6144]
```

to

```
v.customize ['modifyvm', :id, '--memory', 2048]
```

Do a vagrant destroy and vagrant up

**Comment by Hanan100** [comment]
I know this an issue but it is the max that this laptop can handle. right now I did a vagrant up and I got the xenserver up but the management server installation didn’t complete. I got the following error message: 

 management: SSH username: vagrant
    management: SSH auth method: private key
    management: Warning: Connection timeout. Retrying...
==> management: Machine booted and ready!
==> management: Checking for guest additions in VM...
==> management: Configuring and enabling network interfaces...
==> management: Mounting shared folders...
    management: /vagrant => /home/blue/Thesis/devcloud4-master/binary-installation-basic
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

/sbin/mount.vboxsf: mounting failed with the error: No such device

**Comment by imduffy15** [comment]
That's an issue with your host operating system and virtualbox please refer to the virtualbox documentation for installation on ubuntu and loading of the virtualbox kernel modules.

**Comment by imduffy15** [comment]
Hi @Hanan100 did you manage to make progress on this?

**Comment by Hanan100** [comment]
Hi @imduffy15 , No still there is no progress. I don't understand the cause of the issue as it was working fine before, I am trying to trace my steps back to see what did I do differently.

**Comment by imduffy15** [comment]
Have you done any kernel updates that would of affected dkms or the vboxsf
modules? You can try manually load it with modprobe

**Comment by Hanan100** [comment]
I don't recall I have done so but as you said I will try to manually load it with modprobe. But do you think this is also affecting the advanced installation (Not the binary installation advanced) in terms of locating the cloudstack cookbook. As this mainly what I receive as an error.

**Comment by Hanan100** [comment]
Hi @imduffy15 still the error message not resolved but I noticed the following message with regard to GuestAdditions: 
==> management: Machine booted and ready!
GuestAdditions versions on your host (4.3.6) and guest (4.3.8) do not match.

I got the iso image of GuestAdditions versions on your host (4.3.6) and reloaded vagrant but even though it says that the GuestAdditions is installed but it is not running. 

Any ideas ???

**Comment by imduffy15** [comment]
That's just a warning, its not critical. It should allow you to proceed safely. i.e. ignore it.

**Comment by imduffy15** [comment]
So what happens when you ignore this warning and the boot up process continues. Does chef kick in?

**Comment by Hanan100** [comment]
No it doesn't it still present the same error message

Sent from Samsung Mobile

<div>-------- Original message --------</div><div>From: Ian Duffy <notifications@github.com> </div><div>Date:17/01/2015  9:51 PM  (GMT+04:00) </div><div>To: imduffy15/devcloud4 <devcloud4@noreply.github.com> </div><div>Cc: Hanan100 <hanan_alobaidli@hotmail.com> </div><div>Subject: Re: [devcloud4] issue with managment server installation in using advanced folder in devcloud4 (#11) </div><div>
</div>
So what happens when you ignore this warning and the boot up process continues. Does chef kick in?

---

**Comment by imduffy15** [comment]
Same error being cookbook missing or vboxsf?

**Comment by Hanan100** [comment]
Now I am only getting the vboxsf
When I use modprobe  I get a massage that vboxsf  module is not found but the dkms  is installed and up to date.


Sent from Samsung Mobile

<div>-------- Original message --------</div><div>From: Ian Duffy <notifications@github.com> </div><div>Date:17/01/2015  9:58 PM  (GMT+04:00) </div><div>To: imduffy15/devcloud4 <devcloud4@noreply.github.com> </div><div>Cc: Hanan100 <hanan_alobaidli@hotmail.com> </div><div>Subject: Re: [devcloud4] issue with managment server installation in using advanced folder in devcloud4 (#11) </div><div>
</div>
Same error being cookbook missing or vboxsf?
On 17 Jan 2015 17:57, "Hanan100" <notifications@github.com> wrote:

>
> No it doesn't it still present the same error message
>
> Sent from Samsung Mobile
>
> <div>-------- Original message --------</div><div>From: Ian Duffy <
> notifications@github.com> </div><div>Date:17/01/2015 9:51 PM (GMT+04:00)
> </div><div>To: imduffy15/devcloud4 <devcloud4@noreply.github.com>
> </div><div>Cc: Hanan100 <hanan_alobaidli@hotmail.com> </div><div>Subject:
> Re: [devcloud4] issue with managment server installation in using advanced
> folder in devcloud4 (#11) </div><div>
> </div>
> So what happens when you ignore this warning and the boot up process
> continues. Does chef kick in?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/imduffy15/devcloud4/issues/11#issuecomment-70376496
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/imduffy15/devcloud4/issues/11#issuecomment-70376781>.
>

---

**Comment by imduffy15** [comment]
There's definitely something odd with your machine. I just did a clean installation of Ubuntu 14.04.01 within a virtual machine. I installed virtualbox and git from apt, I installed the latest vagrant using the deb from their site and installed the latest chefdk using the deb.

Did a vagrant plugin install vagrant-omnibus and vagrant plugin install vagrant-berkshelf
Did a vagrant up and everything worked as expected.....

As I've no idea whats going on with your machine, a possible quick fix here might be to do a removal and reinstallation of virtualbox.

**Comment by Hanan100** [comment]
I did remove virtual box but it is no use. I am now considering to do a clean install of the os and see if that fixes this thing.


Sent from Samsung Mobile

<div>-------- Original message --------</div><div>From: Ian Duffy <notifications@github.com> </div><div>Date:17/01/2015  10:43 PM  (GMT+04:00) </div><div>To: imduffy15/devcloud4 <devcloud4@noreply.github.com> </div><div>Cc: Hanan100 <hanan_alobaidli@hotmail.com> </div><div>Subject: Re: [devcloud4] issue with managment server installation in using advanced folder in devcloud4 (#11) </div><div>
</div>
There's definitely something odd with your machine. I just did a clean installation of Ubuntu 14.04.01 within a virtual machine. I installed virtualbox and git from apt, I installed the latest vagrant using the deb from their site and installed the latest chefdk using the deb.

Did a vagrant plugin install vagrant-omnibus and vagrant plugin install vagrant-berkshelf
Did a vagrant up and everything worked as expected.....

As I've no idea whats going on with your machine, a possible quick fix here might be to do a removal and reinstallation of virtualbox.

---

**Comment by Hanan100** [comment]
Hi @imduffy15  I've reinstalled the OS on my laptop, my enviroment is as follow: 
OS: ubuntu 12.04 LTS
berks -v  : 3.2.1
vagrant -v: Vagrant 1.7.2
vagrant plugin list
vagrant-berkshelf (4.0.2)
vagrant-librarian-chef (0.2.1)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.3, system)
Virtualbox V4.3.6r91406

As a refrence for the instllation I am using the following pages 
https://docs.chef.io/install_dk.html
http://berkshelf.com/

I downloaded chefdk from : https://downloads.chef.io/chef-dk/ubuntu/#/
After downloading and installing chefdk I set the PATH as follow: 
PATH=$HOME/.chefdk/gem/ruby/2.1.0/bin:/opt/chefdk/bin:$PATH

Right now after doing what was mentioned above I am agetting the following error message when running vagrant up through binary installation basic
==> management: [2015-01-18T10:41:48+00:00] INFO: template[/etc/sysconfig/nfs] sending restart action to service[nfs] (delayed)
==> management: [2015-01-18T10:41:50+00:00] INFO: service[nfs] restarted
==> management: [2015-01-18T10:41:50+00:00] ERROR: Running exception handlers
==> management: [2015-01-18T10:41:50+00:00] ERROR: Exception handlers complete
==> management: [2015-01-18T10:41:50+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> management: [2015-01-18T10:41:51+00:00] ERROR: package[cloudstack-management] (cloudstack::management_server line 25) had an error: Timeout::Error: execution expired
==> management: [2015-01-18T10:41:51+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

And When running vagrant up in advanced (Not the binary instllation advanced) I am getting the following error message
==> management: Running chef-solo...
==> management: [2015-01-18T09:38:08+00:00] INFO: Forking chef instance to converge...
==> management: [2015-01-18T09:38:08+00:00] INFO: *** Chef 12.0.3 ***
==> management: [2015-01-18T09:38:08+00:00] INFO: Chef-client pid: 3250
==> management: [2015-01-18T09:38:10+00:00] INFO: Setting the run_list to ["recipe[cloudstack::default]"] from CLI options
==> management: [2015-01-18T09:38:10+00:00] INFO: Run List is [recipe[cloudstack::default]]
==> management: [2015-01-18T09:38:10+00:00] INFO: Run List expands to [cloudstack::default]
==> management: [2015-01-18T09:38:10+00:00] INFO: Starting Chef Run for localhost
==> management: [2015-01-18T09:38:10+00:00] INFO: Running start handlers
==> management: [2015-01-18T09:38:10+00:00] INFO: Start handlers complete.
==> management: [2015-01-18T09:38:10+00:00] ERROR: Running exception handlers
==> management: [2015-01-18T09:38:10+00:00] ERROR: Exception handlers complete
==> management: [2015-01-18T09:38:10+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> management: [2015-01-18T09:38:10+00:00] ERROR: Cookbook cloudstack not found. If you're loading cloudstack from another cookbook, make sure you configure the dependency in your metadata
==> management: [2015-01-18T09:38:10+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.


So what did I do wrong here????????

**Comment by imduffy15** [comment]
Timeout::Error: execution expired suggests an issue with downloading the RPMs for cloudstack management server. What is your internet connection like?

Retry binary-install-basic it may work second time round.

**Comment by Hanan100** [comment]
I did that but still getting this error message

==> management: [2015-01-18T12:09:03+00:00] INFO: Running queued delayed notifications before re-raising exception
==> management: [2015-01-18T12:09:03+00:00] INFO: template[/etc/sysconfig/nfs] sending restart action to service[nfs] (delayed)
==> management: [2015-01-18T12:09:05+00:00] INFO: service[nfs] restarted
==> management: [2015-01-18T12:09:05+00:00] ERROR: Running exception handlers
==> management: [2015-01-18T12:09:05+00:00] ERROR: Exception handlers complete
==> management: [2015-01-18T12:09:05+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> management: [2015-01-18T12:09:05+00:00] ERROR: package[cloudstack-management] (cloudstack::management_server line 25) had an error: Timeout::Error: execution expired
==> management: [2015-01-18T12:09:05+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

**Comment by imduffy15** [comment]
What's your internet connection like? Are you able to download things from
the shapeblue repos defined in chef_configuration.json without issue?

**Comment by Hanan100** [comment]
I dont have any issue with my internet connection, I was able to download things from the shapeblue repos defined in chef_configuration.json without issue . I thought it might be a problem caused by NTP Synchronization  source of the information (https://docs.chef.io/errors.html). I got NTP installed and configured but this didnt make anychange. Dont get the cause of this issue. 

What I really need to do is to get cloudstack managment server running on my host using eclipse hence i need  advanced insted of binary instllation. But with advanced the cloudstack cookbook is not identified.
